src/components/Main/Schedule/services.ts

```diff
@@ -0,0 +1,114 @@
+// export function
+import { DayIndexOptions } from '../../../shared/types';
+
+const idxToMonth = [
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'June',
+  'July',
+  'Aug',
+  'Sept',
+  'Oct',
+  'Nov',
+  'Dec',
+];
+
+const dayIndex: DayIndexOptions = {
+  Sunday: 0,
+  Monday: 1,
+  Tuesday: 2,
+  Wednesday: 3,
+  Thursday: 4,
+  Friday: 5,
+  Saturday: 6,
+};
+
+const readableDate = (unreadableDate: Date) =>
+  `${
+    unreadableDate.getMonth() + 1
+  }/${unreadableDate.getDate()}/${unreadableDate.getFullYear()}`;
+
+const incrementDay = (date: Date) => new Date(date.setDate(date.getDate() + 1));
+
+function determineStartDate(startDate: string, day: number) {
+  let current = incrementDay(new Date(startDate));
+  while (current.getDay() !== day) current = incrementDay(current);
+  return current;
+}
+
+export function everyBeepDayBetweenTwoDates(
```

Comment:
beep day?

Reply:
hahahahahaha to be honest I don't know how to name this. it's supposed to be any "Friday" or "Saturday, your choice of day between two dates. Naming it "everyWeekDay" didn't make sense because that could mean every SINGLE day. Also, this way you can say our code has character and humor in it.

Reply:
hah. Maybe add a comment then
```js
export function everyBeepDayBetweenTwoDates( // your choice of weekday to repeat between two dates
```
or
```suggestion
export function everyRepeatingDayBetweenTwoDates(
```

Reply:
great suggestion!